suite-common/wallet-utils/src/__tests__/accountUtils.test.ts

```diff
@@ -337,4 +340,114 @@ describe('account utils', () => {
+        expect(isDebugOnlyAccountType('coinjoin', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('taproot', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('ledger', 'btc')).toBe(false);
+        expect(isDebugOnlyAccountType('legacy', 'eth')).toBe(true);
+        expect(isDebugOnlyAccountType('ledger', 'eth')).toBe(true);
+    });
+
+    const accounts = [
+        getWalletAccount({ symbol: 'eth', accountType: 'legacy' }),
+        getWalletAccount({ symbol: 'eth', accountType: 'normal' }),
+        getWalletAccount({ symbol: 'eth', accountType: 'ledger' }),
+        getWalletAccount({ symbol: 'btc', accountType: 'coinjoin' }),
+        getWalletAccount({ symbol: 'btc', accountType: 'taproot' }),
+        getWalletAccount({ symbol: 'btc', accountType: 'legacy' }),
+        getWalletAccount({ symbol: 'btc', accountType: 'segwit' }),
+        getWalletAccount({ symbol: 'btc', accountType: 'ledger' }),
+        getWalletAccount({ symbol: 'matic', accountType: 'legacy' }),
+        getWalletAccount({ symbol: 'matic', accountType: 'normal' }),
+        getWalletAccount({ symbol: 'matic', accountType: 'ledger' }),
+        getWalletAccount({ symbol: 'sol', accountType: 'normal', empty: true, visible: false }),
+        getWalletAccount({ symbol: 'sol', accountType: 'ledger' }),
+        getWalletAccount({
+            symbol: 'sol',
+            accountType: 'ledger',
+            empty: true,
+            visible: false,
+        }),
+    ];
+
+    type RunFilterReceiveAccountsTestParams = {
+        isDebug?: boolean;
+        receiveNetwork?: string;
+        deviceState?: `${string}@${string}:${number}`;
+    };
+
+    const runFilterReceiveAccouns = ({
+        isDebug = true,
+        receiveNetwork = 'eth',
+        deviceState = '1stTestnetAddress@device_id:0',
+    }: RunFilterReceiveAccountsTestParams) => {
+        const device = getSuiteDevice({
+            unavailableCapabilities: {
+                dash: 'no-support',
+            },
+            state: deviceState,
+        });
+        const unavailableCapabilities = device?.unavailableCapabilities ?? {};
+
+        const receiveNetworks = networksCollection.filter(
+            (n: Network) =>
+                n.symbol === receiveNetwork &&
+                !unavailableCapabilities[n.symbol] &&
+                ((n.isDebugOnlyNetwork && isDebug) || !n.isDebugOnlyNetwork),
+        );
+
+        return filterReceiveAccounts({
+            accounts,
+            deviceState: device.state,
+            receiveNetwork,
+            isDebug,
+            receiveNetworks,
+        });
+    };
+
+    it('filterReceiveAccounts: isDebug', () => {
+        const filteredAccounts = [
+            getWalletAccount({ symbol: 'eth', accountType: 'legacy' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'normal' }),
+            getWalletAccount({ symbol: 'eth', accountType: 'ledger' }),
+        ];
```

Comment:
Could add an actually debug account here, so you can test that it has been _filtered out_.

:thought_balloon: tip to make your life easier: you can rename this `filteredAccounts` to `normalAccounts`, then
`const filteredAccounts = [...normalAccounts, getWalletAccount(regtest or something)]`,
and test against `normalAccounts`

Reply:
Eth legacy and ledger accounts are debug accounts

Reply:
regtest is debug network
eth legacy is debug account type